compiler/packages/babel-plugin-react-compiler/src/HIR/HIR.ts

```diff
@@ -1205,6 +1205,20 @@ export type ValidIdentifierName = string & {
   [opaqueValidIdentifierName]: 'ValidIdentifierName';
 };
 
+export function makeTemporary(
+  id: IdentifierId,
+  loc: SourceLocation,
+): Identifier {
+  return {
+    id,
+    name: null,
+    mutableRange: {start: makeInstructionId(0), end: makeInstructionId(0)},
+    scope: null,
+    type: makeType(),
+    loc,
+  };
+}
+
```

Comment:
just a heads up that we also have [`createTemporaryPlace()`](https://github.com/facebook/react/blob/2daa703236019185deceadf747c94f4742a68a12/compiler/packages/babel-plugin-react-compiler/src/HIR/HIRBuilder.ts#L888) which we could maybe unify

Reply:
will do this in a follow up!

Reply:
Done here https://github.com/facebook/react/pull/30585